src/openms/source/METADATA/ID/IdentificationDataConverter.cpp

```diff
@@ -1214,4 +1214,169 @@ namespace OpenMS
     }
   }
 
+
+  void IdentificationDataConverter::importConsensusIDs(ConsensusMap& consensus,
+                                                       bool clear_original)
+  {
+    // collect all peptide IDs:
+    vector<PeptideIdentification> peptides = consensus.getUnassignedPeptideIdentifications();
+    // get peptide IDs from each consensus feature, add meta values:
+    Size id_counter = 0;
+    for (Size i = 0; i < consensus.size(); ++i)
+    {
+      ConsensusFeature& feature = consensus[i];
+      for (const PeptideIdentification& pep : feature.getPeptideIdentifications())
+      {
+        peptides.push_back(pep);
+        // store feature index so we can map the converted ID back;
+        // key needs to be unique in case the same ID matches multiple features:
+        String key = "IDConverter_trace_" + String(id_counter);
+        for (PeptideHit& hit : peptides.back().getHits())
+        {
+          hit.setMetaValue(key, i);
+        }
+        ++id_counter;
+      }
+      if (clear_original) feature.getPeptideIdentifications().clear();
+    }
+
+    IdentificationData& id_data = consensus.getIdentificationData();
+    importIDs(id_data, consensus.getProteinIdentifications(), peptides);
+
+    // map converted IDs back to consensus features using meta values assigned above:
+    for (ID::ObservationMatchRef ref = id_data.getObservationMatches().begin();
+         ref != id_data.getObservationMatches().end(); ++ref)
+    {
+      vector<String> meta_keys;
+      ref->getKeys(meta_keys);
+      for (const String& key : meta_keys)
+      {
+        if (key.hasPrefix("IDConverter_trace_"))
+        {
+          Size index = ref->getMetaValue(key);
+          ConsensusFeature* feat_ptr = &consensus.at(index);
```

Comment:
does this also work?
```suggestion
          ConsensusFeature& feat = consensus.at(index);
```

Reply:
Yep. I've changed it.